docs/docs/guides/troubleshooting.mdx

```diff
@@ -155,6 +155,86 @@ LogBox.ignoreLogs([
 
 See [the accessibility overview](accessibility) to learn more about Reduced Motion.
 
+### Tried to modify key of an object which has been converted to a Shareable.
+
+**Problem:** This warning is displayed to inform the user that a Shared Value should be used or an object used in a worklet should be accessed more granularly.
+
+#### 1. Not using Shared Values.
+
+You might get this warning when you do something along the lines of:
+
+```js
+const obj = { prop: 1 };
+
+function worklet() {
+  'worklet';
+  console.log(obj.prop);
+}
+
+runOnUI(worklet)();
+obj.prop = 2; // Warning: Tried to modify key `prop` of an object which has been already passed to a worklet.
+runOnUI(worklet)();
+```
+
+and expect the results to be `1` and `2`. However, the results will be `1` and `1` because `obj` is not a Shared Value and is only copied to UI runtime once. Therefore, in development builds, we make the object immutable and add this warning after copying it to signal that it's not a valid use of Reanimated. To fix this, you should use a Shared Value instead:
+
+**Solution:**
+
+```diff
+-const obj = { prop: 1 };
++const sv = useSharedValue({ prop: 1 });
+
+ function worklet() {
+   'worklet';
+-  console.log(obj.prop);
++  console.log(sv.value.prop);
+ }
+
+ runOnUI(worklet)();
+-obj.prop = 2; // Warning: Tried to modify key `prop` of an object which has been already passed to a worklet.
++sv.value = { prop: 2 }; // Everything is fine here.
++// Keep in mind that you cannot modify the property directly with `sv.value.prop = 2`
```

Comment:
You can also use `.modify` here but usually it's not recommended.

Reply:
Actually let's add it.